app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.ciandt.dragonfly.example.features.modelselection
+
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.LinearSnapHelper
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.SimpleItemAnimator
+import android.widget.Button
+import com.ciandt.dragonfly.data.Model
+import com.ciandt.dragonfly.example.BaseActivity
+import com.ciandt.dragonfly.example.MainActivity
+import com.ciandt.dragonfly.example.R
+
+
+class ModelSelectionActivity : BaseActivity(), ModelSelectionContract.View {
+
+    private val MODELS_BUNDLE = "MODELS_BUNDLE"
+
+    private lateinit var presenter: ModelSelectionContract.Presenter
+
+    private lateinit var recyclerView: RecyclerView
+    private var models = ArrayList<Model>()
+
+    private lateinit var aboutButton: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_model_selection)
+
+        presenter = ModelSelectionPresenter(this)
```

Comment:
I saw some reference projects proposing to presenter.attach(this) onResume() and presenter.detach() (sets view reference to null internally) onPause().

I suppose the reference to the view should be cleaned, otherwise we may run into memory leaks on configuration changes, for example, no?